### PR TITLE
feat: update docs for new query server config

### PR DESCRIPTION
**Pull request by janl**
See https://github.com/apache/couchdb/pull/1602

### CODE REVIEW

**Comment by wohali** [comment]
Think this needs updating for changes to SSL config still. We fully scrubbed out proxy stuff and os_daemons already.

Don't think we need to document configuring the http global handlers, but if you want, go for it.

**Comment by wohali** [comment]
@janl since I merged the release notes already, and it includes everything for this PR, just drop your version of `src/whatsnew/2.3.rst` and we should be fine.

**Comment by janl** [comment]
@wohali can you take this over from me?

**Comment by wohali** [comment]
@Janl ok

**Comment by wohali** [comment]
@flimzy can I get a final review here? I've rewritten a bunch of the...problematic text surrounding these changes to be more up to date (for instance, replacing "SSL config" with "HTTPS config", which I think more people will understand.)

File: src/config/query-servers.rst [hunk 1]
```diff
@@ -53,26 +53,28 @@ Query Servers Definition
...
 
     For more info about the available options, please consult ``couchjs -h``.
 
     .. _Mozilla SpiderMonkey: https://developer.mozilla.org/en/docs/SpiderMonkey
 
     .. seealso::
-        :ref:`Native Erlang Query Server <config/native_query_servers>` that
-        allows to process Erlang `ddocs` and runs within CouchDB bypassing
+        The :ref:`Mango Query Server <api/db/_find>` is a declarative language
+        for indexing and finding data in CouchDB that requires *no programming*.
```
Comment (flimzy) [inline src/config/query-servers.rst]:
```suggestion
        and is used for indexing and finding data in CouchDB. 
```

File: src/config/query-servers.rst [hunk 2]
```diff
@@ -53,26 +53,28 @@ Query Servers Definition
...
+        COUCHDB_QUERY_SERVER_JAVASCRIPT="/usr/bin/couchjs -S 536870912 /usr/share/server/main.js"
 
     For more info about the available options, please consult ``couchjs -h``.
 
     .. _Mozilla SpiderMonkey: https://developer.mozilla.org/en/docs/SpiderMonkey
 
     .. seealso::
-        :ref:`Native Erlang Query Server <config/native_query_servers>` that
-        allows to process Erlang `ddocs` and runs within CouchDB bypassing
+        The :ref:`Mango Query Server <api/db/_find>` is a declarative language
```
Comment (flimzy) [inline src/config/query-servers.rst]: The language, not the data, requires no programming :) 

```suggestion
        The :ref:`Mango Query Server <api/db/_find>` is a declarative language that requires *no programming*, 
```

**Comment by wohali** [comment]
@flimzy Good catch, fixed! Will consider that a +1 and merge momentarily.

Thanks again for doing the bulk of the work here @janl !

**Pull request merged**